### PR TITLE
HDDS-11469. Statistics of Pipeline and Container

### PR DESCRIPTION
**Pull request by jianghuazhu**
## What changes were proposed in this pull request?
In SCM, a lot of Pipeline and Container data are collected. The purpose of this pr is to display them more intuitively.
This pr modifies some UI and jmx.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-11469

## How was this patch tested?
ci: https://github.com/jianghuazhu/ozone/actions/runs/10927019250
New SCM ui:
image
image

### CODE REVIEW

**Comment by jianghuazhu** [comment]
@ivandika3 @devabhishekpal @ArafatKhan2198 , can you help review this PR?
Thanks.

**Comment by jianghuazhu** [comment]
@adoroszlai @ivandika3 @devabhishekpal @ArafatKhan2198  , can you take a look at this PR?
Thanks.

File: hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm.js [hunk 1]
```diff
@@ -52,6 +52,32 @@
...
+                            under_replicated : "N/A",
+                            mis_replicated : "N/A",
+                            over_replicated : "N/A",
+                            missing : "N/A",
+                            unhealthy : "N/A",
+                            empty : "N/A",
+                            open_unhealthy : "N/A",
+                            quasi_closed_stuck : "N/A",
+                            open_without_pipeline : "N/A"
+                        }
```
Comment (ivandika3) [inline hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm.js:80]: Could you flatten the hierarchy for different types of metrics? In my opinion, pipelines and containers do not "belong" to "nodes" so they should not be nested under it.

I suggest something like:

- datanode: `statistics.nodes`
- pipeline: `statistics.pipelines`
- containers: `statistics.containers`

File: hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm.js [hunk 2]
```diff
@@ -142,6 +168,46 @@
...
+                    $scope.statistics.nodes.data.container.under_replicated = ctrl.scmcontainermanager.UnderReplicatedContainers;
+                    $scope.statistics.nodes.data.container.mis_replicated = ctrl.scmcontainermanager.MisReplicatedContainers;
+                    $scope.statistics.nodes.data.container.over_replicated = ctrl.scmcontainermanager.OverReplicatedContainers;
+                    $scope.statistics.nodes.data.container.missing = ctrl.scmcontainermanager.MissingContainers;
+                    $scope.statistics.nodes.data.container.unhealthy = ctrl.scmcontainermanager.UnhealthyContainers;
+                    $scope.statistics.nodes.data.container.empty = ctrl.scmcontainermanager.EmptyContainers;
+                    $scope.statistics.nodes.data.container.open_unhealthy = ctrl.scmcontainermanager.OpenUnhealthyContainers;
+                    $scope.statistics.nodes.data.container.quasi_closed_stuck = ctrl.scmcontainermanager.StuckQuasiClosedContainers;
+                    $scope.statistics.nodes.data.container.open_without_pipeline = ctrl.scmcontainermanager.OpenContainersWithoutPipeline;
```
Comment (ivandika3) [inline hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm.js:208]: Let's separate the prefix between the container lifecycle and health metrics:

- Container lifecycle metrics: `statistics.container.lifecycle`
- Health metrics: `statistics.container.health`

File: hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm-overview.html [hunk 3]
```diff
@@ -110,6 +110,106 @@ <h2>Space Statistics</h2>
...
+    </tr>
+    <tr>
+        <td>Dormant</td>
+        <td>{{statistics.nodes.data.pipeline.dormant}}</td>
+    </tr>
+    </tbody>
+</table>
+
+<h2>Container Statistics</h2>
+<table class="table table-bordered table-striped">
```
Comment (ivandika3) [inline hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm-overview.html:140]: Let's separate this table into two smaller ones: one for "Lifecycle State" and "Health State"

**Review by ivandika3 — COMMENTED** [review]
@jianghuazhu Thanks for the patch. Left a few comments.

File: hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm-overview.html (hunk 3 above)
Comment (devabhishekpal) [reply hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm-overview.html:140]: I think we could shorten this to just Operational State and Health like how we do in Recon?

Comment (ivandika3) [reply hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm-overview.html:140]: Sure, let's make it consistent with Recon.

File: hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm-overview.html [hunk 4]
```diff
@@ -110,6 +110,116 @@ <h2>Space Statistics</h2>
...
+        <td>{{statistics.pipelines.open}}</td>
+    </tr>
+    <tr>
+        <td>Dormant</td>
+        <td>{{statistics.pipelines.dormant}}</td>
+    </tr>
+    </tbody>
+</table>
+
+<h2>Container Lifecycle Statistics</h2>
```
Comment (ivandika3) [inline hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm-overview.html:139]: I think the tables can be put under one `h2` header.

```
<h2>Container Statistics</h2>
<table class="table table-bordered table-striped">
    <tbody>
    <tr>
        <th>Operational State</th>
        <th>Size</th>
    </tr>
    ...
</table>
<table class="table table-bordered table-striped">
    <tbody>
    <tr>
        <th>Health</th>
        <th>Size</th>
    </tr>
    ...
</table>
```

**Review by ivandika3 — COMMENTED** [review]
@jianghuazhu Thanks for the update. One minor comment, other than that LGTM.

Could you also update the screenshot with the latest changes? Thank you.

File: hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm-overview.html (hunk 4 above)
Comment (jianghuazhu) [reply hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm-overview.html]: Thanks @ivandika3 .
I have updated it.
Here is the latest UI:
image

And I have updated the description of the PR.

Comment (ivandika3) [reply hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm-overview.html]: Thanks for the update, is it possible to have a white space in between the table?

Comment (jianghuazhu) [reply hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm-overview.html]: @ivandika3 , this is the UI style we expect, right?
image

Comment (ivandika3) [reply hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm-overview.html]: Yes, that's what I expected, but the previous screenshot did not have a white space between the table.

Comment (jianghuazhu) [reply hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm-overview.html]: Thanks @ivandika3 .
I have updated it.
And I have updated the description of this PR.

Comment (ivandika3) [reply hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm-overview.html]: Thanks it looks good.

**Review by ivandika3 — APPROVED** [review]
@jianghuazhu Thanks for the update. LGTM +1.

**Comment by ivandika3** [comment]
Thanks @jianghuazhu for the patch, and @devabhishekpal for the review.
